src/main/java/org/folio/rest/impl/PrecedingSucceedingTitleAPI.java

```diff
@@ -68,6 +94,56 @@ public void putPrecedingSucceedingTitlesByPrecedingSucceedingTitleId(String prec
       PutPrecedingSucceedingTitlesByPrecedingSucceedingTitleIdResponse.class, asyncResultHandler);
   }
 
+  private boolean validatePrecedingSucceedingTitles(List<PrecedingSucceedingTitle> precedingSucceedingTitles,
+                                                    String instanceId,
+                                                    Handler<AsyncResult<Response>> asyncResultHandler) {
+    boolean areValidTitles = true;
+    for (PrecedingSucceedingTitle precedingSucceedingTitle : precedingSucceedingTitles) {
+      if (titleIsLinkedToInstanceId(precedingSucceedingTitle, instanceId)) {
+        var validationErrorMessage =
+          createValidationErrorMessage("precedingInstanceId or succeedingInstanceId", "",
+            String.format("The precedingInstanceId or succeedingInstanceId should contain instanceId [%s]", instanceId));
+        asyncResultHandler.handle(Future.succeededFuture(PutPrecedingSucceedingTitlesResponse
+          .respond422WithApplicationJson(validationErrorMessage)));
+        areValidTitles = false;
+      }
+    }
+    return areValidTitles;
+  }
+
+  private boolean titleIsLinkedToInstanceId(PrecedingSucceedingTitle precedingSucceedingTitle, String instanceId) {
+    return !instanceId.equals(precedingSucceedingTitle.getPrecedingInstanceId())
+      && !instanceId.equals(precedingSucceedingTitle.getSucceedingInstanceId());
+  }
+
+  private Future<Response> saveCollection(List<PrecedingSucceedingTitle> entities,
+                                          Map<String, String> okapiHeaders, Context vertxContext) {
+    Promise<Response> promise = Promise.promise();
+
+    try {
+      MetadataUtil.populateMetadata(entities, okapiHeaders);
+      PostgresClient postgresClient = postgresClient(vertxContext, okapiHeaders);
+      Handler<AsyncResult<RowSet<Row>>> replyHandler = result -> {
+        if (result.failed()) {
+          var errorMessage = result.cause().getMessage();
+          PutPrecedingSucceedingTitlesResponse response;
+          if (ValidationHelper.isFKViolation(errorMessage)) {
+            response = PutPrecedingSucceedingTitlesResponse.respond404WithTextPlain("Instance not found");
+          } else {
+            response = PutPrecedingSucceedingTitlesResponse.respond500WithTextPlain(errorMessage);
+          }
+          promise.complete(response);
+        } else {
+          promise.complete(PutPrecedingSucceedingTitlesResponse.respond204());
+        }
+      };
+      postgresClient.saveBatch(PRECEDING_SUCCEEDING_TITLE_TABLE, entities, replyHandler);
+    } catch (ReflectiveOperationException e) {
+      promise.complete(PutPrecedingSucceedingTitlesResponse.respond500WithTextPlain(e.getMessage()));
+    }
+    return promise.future();
```

Comment:
[Futurisation](https://github.com/folio-org/raml-module-builder/blob/master/doc/futurisation.md) allows to greatly reduce the boilerplate code:

```suggestion
  private Future<Response> saveCollection(List<PrecedingSucceedingTitle> entities,
                                          Map<String, String> okapiHeaders, Context vertxContext) {
    try {
      MetadataUtil.populateMetadata(entities, okapiHeaders);
      return postgresClient(vertxContext, okapiHeaders)
          .saveBatch(PRECEDING_SUCCEEDING_TITLE_TABLE, entities)
          .<Response>map(x -> PutPrecedingSucceedingTitlesResponse.respond204())
          .otherwise(cause -> {
            var errorMessage = cause.getMessage();
            if (ValidationHelper.isFKViolation(errorMessage)) {
              return PutPrecedingSucceedingTitlesResponse.respond404WithTextPlain("Instance not found");
            } else {
              return PutPrecedingSucceedingTitlesResponse.respond500WithTextPlain(errorMessage);
            }
          });
    } catch (Throwable e) {
      return Future.succeededFuture(PutPrecedingSucceedingTitlesResponse.respond500WithTextPlain(e.getMessage()));
    }
  }
```

Reply:
This is really better. Thanks.